psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -418,12 +428,83 @@ void MintsHelper::integral_hessians()
     throw FeatureNotImplemented("libmints", "MintsHelper::integral_hessians", __FILE__, __LINE__);
 }
 
+void MintsHelper::one_body_ao_computer(std::vector<std::shared_ptr<OneBodyAOInt>> ints,
+                                       SharedMatrix out, bool symm) {
+    // Grab basis info
+    std::shared_ptr<BasisSet> bs1 = ints[0]->basis1();
+    std::shared_ptr<BasisSet> bs2 = ints[0]->basis2();
+
+    // Limit to the number of incoming onbody ints
+    size_t nthread = nthread_;
+    if (nthread > ints.size()) {
+        nthread = ints.size();
+    }
+
+    // Grab the buffers
+    std::vector<const double *> ints_buff(nthread);
+    for (size_t thread = 0; thread < nthread; thread++) {
+        ints_buff[thread] = ints[thread]->buffer();
+    }
+
+    double **outp = out->pointer();
+
+    // Loop it
+    #pragma omp parallel for schedule(guided) num_threads(nthread)
+    for (size_t MU = 0; MU < bs1->nshell(); ++MU) {
+        const size_t num_mu = bs1->shell(MU).nfunction();
+        const size_t index_mu = bs1->shell(MU).function_index();
+
+        size_t rank = 0;
+#ifdef _OPENMP
+        rank = omp_get_thread_num();
```

Comment:
Thats just asking which thread it is. The one to worry about is `omp_get_max_threads`.